init-mqtt.h

```diff
@@ -103,27 +107,84 @@ bool connectMqtt() {
   return res;
 }
 
-void initMqtt() {
+bool reconnectMqtt() {
+  return connectMqtt(mqttConfig.user, mqttConfig.pwd);
+}
 
-  mqttClient.setCallback(mqttCallback);
-  
+void setMqttServer(const char* server, int port){
+  delay(500);
+  mqttClient.setServer(server, port);
+  Serial.print("Mqtt server: ");  
+  Serial.print(server);
+  Serial.print(":");
+  Serial.println(port); 
+}
+
+void initMqtt() {
+  mqttClient.setCallback(mqttCallback);  
   loadMqttConfig();
-  //temporary mqtt configs. for testing...
-  //setMqttConfig("m23.cloudmqtt.com", 12925,"tlwhlgqr", "g-VQc5c6w7eN");
-  setMqttConfig("m13.cloudmqtt.com", 12730,"wnezdvgh", "ycm1xiuzX936");
-  if (isMqttConfigSet) {
-    mqttClient.setServer(mqttConfig.server, mqttConfig.port);    
-  }
+  if(isMqttConfigSet){
+    setMqttServer(mqttConfig.server, mqttConfig.port);
+  }  
 }
 
 void publishInt(const char* topic, int d) {
   char payload[5];
   sprintf(payload, "%d", d);
-  Serial.print("Topic: ");
-  Serial.println(topic);
-  Serial.print("Payload: ");
-  Serial.println(payload);
   mqttClient.publish(topic, payload);
 }
 
+bool tryConnectMqtt(const char* server, int port, const char* user, const char* pwd){
+  bool res = false;
+  setMqttServer(server, port);
+  uint8_t attempts = 0;
+  Serial.println("Connecting Mqtt");
+  while (!connectMqtt(user, pwd) && attempts < MQTT_TIMEOUT) {
+     delay(500);
+     attempts++;
+  }
+
+  if (attempts < MQTT_TIMEOUT) {
+    isWifiConfigSet = true;
+    res = true;
+    Serial.println("MQTT Connected");
+  } else {
+    Serial.println("Failed to connect MQTT");
+  }
+  return res;
+}
+
+uint8_t saveMqttConfig(JsonObject& json) {
```

Comment:
Function name isn`t clear(for me).
Name supposes that this func saves json object to memory(ram and flash). But it also tries to connect to mqtt broker and returns fault codes.

Reply:
will look there

Reply:
the described behaviour is absolutely intended - it tries to connect mqtt with configs provided in json -  and updated them if success

will rename to handleMqttJson